tests/unit/cocoon/test_utilities.py

```diff
@@ -1249,3 +1249,297 @@ def test_checkargs_with_incorrect_type(self, _, function, param):
     def test_checkargs_with_invalid_argument(self, _, function, kwargs):
         with self.assertRaises(ValueError):
             function(**kwargs)
+
+    def test_scale_quote_of_type(self):
```

Comment:
we probably need some more tests here:
- column contains missing/`None` quote values
- no `type_code`s match

Reply:
- add test for None values for not_of type, and of_type
- add test for invalid type & price column names

Reply:
- added test for missing not_of_type values and missing of_type values
- added tests for invalid type and price column mappings